Source/cmMakefileExecutableTargetGenerator.cxx

```diff
@@ -264,10 +265,11 @@ void cmMakefileExecutableTargetGenerator::WriteNvidiaDeviceExecutableRule(
   if (useLinkScript) {
     // Use a link script.
     const char* name = (relink ? "drelink.txt" : "dlink.txt");
-      this->CreateLinkScript(name, real_link_commands, commands1, depends);
-      if(this->Makefile->IsOn("CMAKE_EXPORT_LINK_COMMANDS")){
-        this->CreateLinkScriptJSON(name, real_link_commands, depends);
-      }
+    this->CreateLinkScript(name, real_link_commands, commands1, depends);
```

Comment:
There are two places with this if. Please, consider both. This is first

Reply:
They call in different function. This is WriteNvidiaDeviceExecutableRule function